javatests/arcs/showcase/references/Arcs.kt

```diff
@@ -35,116 +76,77 @@ class Arcs(
 ) {
     val schedulerProvider = JvmSchedulerProvider(EmptyCoroutineContext)
 
-
     private val fakeLifecycleOwner = object : LifecycleOwner {
         private val lifecycle = LifecycleRegistry(this)
         override fun getLifecycle() = lifecycle
     }
 
-
-    val arcHost = ArcHost(
+    val activationFactory = ServiceStoreFactory(
         context,
         fakeLifecycleOwner.lifecycle,
-        schedulerProvider,
-        connectionFactory
+        connectionFactory = connectionFactory
     )
 
-    private lateinit var allocator: Allocator
+    val arcHost = ArcHost(
+        schedulerProvider,
+        activationFactory = activationFactory
+    )
 
-    /**
-     * Start the arc containing the particles that provide the persistence. This should be run
-     * one time per instance, via [startIfNot].
-     */
-    private suspend fun startArc() {
-        /** This should probably live in an initializer somewhere. */
-        val hostRegistry = ExplicitHostRegistry().apply {
+    val hostRegistry = ExplicitHostRegistry().apply {
+        runBlocking {
             registerHost(arcHost)
         }
-
-        allocator = Allocator.create(
-            hostRegistry,
-            EntityHandleManager(
-                arcId = "allocator",
-                hostId = "allocator",
-                time = JvmTime,
-                scheduler = schedulerProvider.invoke("allocator")
-            )
-        )
-
-        allocator.startArcForPlan(WriteRecipePlan)
     }
 
-    suspend fun stop() {
-        allocator.stopArc("!:testArc".toArcId())
-    }
-
-    private var started = false
-    val mutex = Mutex()
-
-    private suspend fun startIfNot() = mutex.withLock {
-        if (!started) {
-            startArc()
-            started = true
-        }
-    }
+    val allocator = Allocator.create(
+        hostRegistry,
+        EntityHandleManager(
+            arcId = "allocator",
+            hostId = "allocator",
+            time = JvmTime,
+            scheduler = schedulerProvider.invoke("allocator")
+        )
+    )
 
-    fun all0(): List<MyLevel0> {
-        return runBlocking {
-            startIfNot()
-            arcHost.reader0.read()
-        }
+    suspend fun <T : Particle> getParticle(
+        plan: Plan,
+        particleName: String
+    ): T {
+        val arc = startArc(plan)
+        return getParticle(arc.id, particleName)
     }
 
-    fun put0(item: MyLevel0) {
-        runBlocking {
-            startIfNot()
-            arcHost.writer0.write(item)
-        }
+    suspend fun startArc(plan: Plan): Arc {
+        val arc = allocator.startArcForPlan(plan)
+        arc.waitForStart()
```

Comment:
Just confirming, is this valid/sufficient now or do we still need to do the workaround with checking all particle handles?

Reply:
Right now, the particles are also checking for handles to be ready. b/158233725 is still open, so I assume this is still required.